### PR TITLE
breaking(Button): replace `type` prop with `secondary` and `primary`

### LINKED ISSUE #374: Button: `type` is working incorrecly
## Bug Report

Hi guys, I'm building now some stuff based on Stardust specifications and I found the design problem.

image

image

However, the `type` prop is reserved in `button`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

### Steps

```jsx
<Button content="Click here" type="reset" />
```

### Expected Result

```html
<button class="ui-button" aria-disabled="false" type="reset">
  <span class="ui-slot">Click here</span>
</button>
```

### Actual Result

```html
<button class="ui-button" aria-disabled="false">
  <span class="ui-slot">Click here</span>
</button>
```

### Version
0.9.1

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #374, please refer there to get more details.

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -32,9 +32,11 @@ export interface ButtonProps {
   iconPosition?: 'before' | 'after'
   onClick?: ComponentEventHandler<ButtonProps>
   onFocus?: ComponentEventHandler<ButtonProps>
+  primary?: boolean
   renderIcon?: ShorthandRenderFunction
   text?: boolean
-  type?: 'primary' | 'secondary'
+  type?: 'type' | 'submit' | 'reset'
```
Comment (bmdalex) [inline src/components/Button/Button.tsx]: can we have 
```jsx
<button type="type">click</button>
```
?

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -105,11 +107,17 @@ class Button extends UIComponent<Extendable<ButtonProps>, ButtonState> {
...
     onFocus: PropTypes.func,
 
+    /** A button can be formatted to show different levels of emphasis. */
+    primary: PropTypes.bool,
+
     /** A button can be formatted to show only text in order to indicate some less-pronounced actions. */
     text: PropTypes.bool,
 
+    /** The type prop specifies the type of button. */
+    type: PropTypes.oneOf(['button', 'submit', 'reset']),
```
Comment (bmdalex) [inline src/components/Button/Button.tsx]: We don't seem to handle `type` so I think we should remove it from `propTypes` and the interface

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -156,6 +164,7 @@ class Button extends UIComponent<Extendable<ButtonProps>, ButtonState> {
         disabled={disabled}
         onClick={this.handleClick}
         onFocus={this.handleFocus}
+        type={type}
```
Comment (bmdalex) [inline src/components/Button/Button.tsx]: if we decide to remove type from handled props don't forget to remove it from here as well

**Review by bmdalex — CHANGES_REQUESTED** [review]
pls take a look at comments

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (layershifter) [reply src/components/Button/Button.tsx]: Nope, it was a typo

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (layershifter) [reply src/components/Button/Button.tsx]: Fully agree, I will remove the `type` prop at all

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -183,16 +187,14 @@ class Button extends UIComponent<Extendable<ButtonProps>, ButtonState> {
...
 
     if (disabled) {
       e.preventDefault()
       return
     }
 
-    if (onClick) {
-      onClick(e, this.props)
-    }
+    _.invoke(this.props, 'onClick', e, this.props)
```
Comment (layershifter) [inline src/components/Button/Button.tsx:197]: Simplify with `invoke()`

**Review by miroslavstastny — COMMENTED** [review]
The added test is failing.

If we decided to remove `type` for `Button` we should do the same for `Menu` and `Divider` in the same release.

**Review by miroslavstastny — APPROVED** [review]
Don't forget to update `CHANGELOG.md`

File: CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ### Fixes
 - Fix endMedia to not be removed from DOM on mouseleave for `ListItem` @musingh1 ([#278](https://github.com/stardust-ui/react/pull/278))
+- Replace the `type` prop with `secondary` and `primary` for `Button` @layershifter ([#419](https://github.com/stardust-ui/react/pull/419))
```
Comment (miroslavstastny) [inline CHANGELOG.md]: This should  go to BREAKING CHANGES!

**Review by miroslavstastny — CHANGES_REQUESTED** [review]

**Review by bmdalex — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (layershifter) [reply CHANGELOG.md]: Nice catch, done

**Comment by bmdalex** [comment]
agreed with @miroslavstastny
@layershifter  
do we have an issue/ticket to track the other `type` props (for `Menu` and `Divider `)?
we should have PRs for these soon, i can help with that
fyi @alinais

**Review by miroslavstastny — APPROVED** [review]

**Pull request merged**